packages/buttons/src/styled/StyledButton.ts

```diff
@@ -319,27 +333,35 @@ const sizeStyles = (props: IButtonProps & ThemeProps<DefaultTheme>) => {
   } else {
     const height = getHeight(props);
     const lineHeight = math(`${height} - (${props.theme.borderWidths.sm} * 2)`);
-    let padding;
     let fontSize;
+    let size;
 
-    if (props.size === 'small') {
-      fontSize = props.theme.fontSizes.sm;
-      padding = `${props.theme.space.base * 3}px`;
+    if (props.size === 'large') {
+      size = 'lg';
+    } else if (props.size === 'small') {
+      size = 'sm';
     } else {
-      fontSize = props.theme.fontSizes.md;
+      size = 'md';
+    }
 
-      if (props.size === 'large') {
-        padding = `${props.theme.space.base * 5}px`;
-      } else {
-        padding = `${props.theme.space.base * 4}px`;
-      }
+    if (props.size === 'small') {
+      fontSize = 'sm';
+    } else {
+      fontSize = 'md';
     }
 
     retVal = css`
-      padding: 0 ${em(math(`${padding} - ${props.theme.borderWidths.sm}`), fontSize)};
-      height: ${height};
+      padding: 0
+        var(
+          --${prefix}--buttons_${size}--padding,
+          ${em(
+            math(`${sizes[size].padding}px - ${props.theme.borderWidths.sm}`),
+            sizes[fontSize].fontSize
+          )}
+        );
+      height: var(--${prefix}--buttons_${size}--height, ${height});
       line-height: ${lineHeight};
-      font-size: ${fontSize};
+      font-size: var(--${prefix}--buttons_${fontSize}--fontSize, ${sizes[fontSize].fontSize});
```

Comment:
It might serve us well to isolate custom property definitions like this in a separate file, so we can reference exactly which properties a component has tied to theming. Especially if we have custom properties not tied directly to specific properties (i.e. shades).

Reply:
Regardless of how that works, it felt natural to have the custom properties take priority should a consumer want to override them from that direction.